File: TicTacToe_Client/Models/Coordinate.cs
using Newtonsoft.Json;

public record Coordinate(
    [JsonProperty("x")] InputPosition x,
    [JsonProperty("y")] InputPosition y
)
{
    public static Coordinate createCoodinate(int xNumber, int yNumber, int max)
    {
        var xInput = new InputPosition(xNumber);
        xInput.assertValid(max);
        var yInput = new InputPosition(yNumber);
        yInput.assertValid(max);
        return new Coordinate(xInput, yInput);
    }
}

File: TicTacToe_Server/app/src/main/java/tictactoe_server/model/Coordinate.java
package tictactoe_server.model;

import tictactoe_server.util.InputPositionValidation;

public record Coordinate(InputPosition x, InputPosition y) {
    public static Coordinate makeCoordinate(int xNumber, int yNumber, int max) {
        InputPosition xIp = new InputPosition(xNumber);
        InputPositionValidation.assertValid(xIp, max);
        InputPosition yIp = new InputPosition(yNumber);
        InputPositionValidation.assertValid(yIp, max);
        return new Coordinate(xIp, yIp);
    }
}